modules/core/src/main/java/org/apache/ignite/internal/processors/performancestatistics/PerformanceStatisticsProcessor.java

```diff
@@ -278,6 +325,32 @@ private void stopWriter() {
         log.info("Performance statistics writer stopped.");
     }
 
+    /** Rotate performance statistics writer. */
+    private void rotateWriter() throws IgniteCheckedException, IOException {
+        try {
+            synchronized (mux) {
+                if (writer != null) {
+                    FilePerformanceStatisticsWriter newWriter = new FilePerformanceStatisticsWriter(ctx);
+
+                    newWriter.start();
+
+                    FilePerformanceStatisticsWriter oldWriter = writer;
+
+                    writer = newWriter;
+
+                    oldWriter.stop();
+                }
+            }
+
+            log.info("Performance statistics writer rotated.");
```

Comment:
Let's log a file that can be used for analysis here.

`Performance statistics writer rotated[writtenFile=001.prf]`

Reply:
done